includes/class-wp-document-revisions-recently-revised-widget.php

```diff
@@ -68,27 +53,63 @@ public function widget_gen( $args, $instance ) {
 		if ( ! $documents ) {
 			return '';
 		}
+		// when getting the images, we may get generated images - getting them will use a cached version of std directory.
 
 		// buffer output to return rather than echo directly.
 		ob_start();
 
 		// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
 		echo $args['before_widget'] . $args['before_title'] . esc_html( apply_filters( 'widget_title', $instance['title'] ) ) . $args['after_title'] . '<ul>';
 
+		if ( (bool) $instance['show_thumb'] ) {
+			// PDF files may have a generated image, and the access call uses a cached version of the (std) upload directory
+			// so cannot change within call and may be wrong, so possibly replace it in the output.
+			$std_dir = str_replace( ABSPATH, '', $wpdr::$wp_default_dir['basedir'] );
+			$doc_dir = str_replace( ABSPATH, '', $wpdr->document_upload_dir() );
+		}
+
 		foreach ( $documents as $document ) {
-			$link = ( current_user_can( 'edit_document', $document->ID ) ) ? add_query_arg(
+			$link   = ( current_user_can( 'edit_document', $document->ID ) ) ? add_query_arg(
 				array(
 					'post'   => $document->ID,
 					'action' => 'edit',
 				),
 				admin_url( 'post.php' )
 			) : get_permalink( $document->ID );
+			$target = ( $instance['new_tab'] ? ' target="_blank"' : '' );
 			// translators: %1$s is the time ago in words, %2$s is the author.
 			$format_string = ( $instance['show_author'] ) ? __( '%1$s ago by %2$s', 'wp-document-revisions' ) : __( '%1$s ago', 'wp-document-revisions' );
 			?>
 			<li>
-				<a href="<?php echo esc_attr( $link ); ?>"><?php echo esc_html( get_the_title( $document->ID ) ); ?></a><br />
-				<?php printf( esc_html( $format_string ), esc_html( human_time_diff( strtotime( $document->post_modified_gmt ) ) ), esc_html( get_the_author_meta( 'display_name', $document->post_author ) ) ); ?>
+				<h3 class="wp-block-post-title"><a href="<?php echo esc_attr( $link ) . '"' . esc_attr( $target ) . '>' . esc_html( get_the_title( $document->ID ) ); ?></a></h3>
+				<?php
+				if ( (bool) $instance['show_thumb'] ) {
+					$thumb = get_post_thumbnail_id( $document->ID );
+					if ( $thumb ) {
+						$thumb_image     = wp_get_attachment_image_src( $thumb, 'medium' );
+						$thumb_image_alt = get_post_meta( $thumb, '_wp_attachment_image_alt', true );
+						// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
```

Comment:
Curious why `thumb_image` and `thumb_image_alt` can't/shouldn't be escaped in this context, if it's going to be an HTML attribute? If `thumb_image_alt` is user controlled, that might allow for XSS.

Reply:
Thanks. Should and will be.